datafusion/core/src/physical_optimizer/sanity_checker.rs

```diff
@@ -125,6 +127,14 @@ pub fn check_plan_sanity(
         plan.required_input_ordering().iter(),
         plan.required_input_distribution().iter()
     ) {
+        // TEMP HACK WORKAROUND https://github.com/apache/datafusion/issues/11492
+        if child.as_any().downcast_ref::<UnionExec>().is_some() {
+            continue;
+        }
+        if child.as_any().downcast_ref::<SortExec>().is_some() {
```

Comment:
Locally i had to disable checking for `SortExec` as well to get our tests passing

Reply:
The likely cause is https://github.com/apache/datafusion/pull/11678